src/main/groovy/com/jakewharton/sdkmanager/internal/PackageResolver.groovy

```diff
@@ -194,6 +193,72 @@ class PackageResolver {
     }
   }
 
+  def resolveEmulator() {
+    def emulatorVersion = project.sdkManager.emulatorVersion
+    if (emulatorVersion == null) {
+      log.debug 'No emulator defined'
+      return
+    }
+
+    def emulatorArchitecture = project.sdkManager.emulatorArchitecture
+    if (emulatorArchitecture == null) {
+      emulatorArchitecture = 'armeabi-v7a'
+      log.debug 'No architecture specified, defaulting to armeabi-v7a'
+    }
+
+    log.debug "Found emulator: $emulatorVersion $emulatorArchitecture"
+
+    def emulatorDir = new File(sdk, FD_SYSTEM_IMAGES + "/$emulatorVersion/$emulatorArchitecture")
+    def alternativeEmulatorDir = new File(sdk, FD_SYSTEM_IMAGES + "/$emulatorVersion/default/$emulatorArchitecture")
+    def emulatorPackage = "sys-img-$emulatorArchitecture-$emulatorVersion"
+    def needsDownload = false
+    if (!folderExists(emulatorDir) && !folderExists(alternativeEmulatorDir)) {
+      needsDownload = true
+      log.lifecycle "Emulator $emulatorVersion $emulatorArchitecture missing. Downloading..."
+    } else {
+      def emulatorPropertiesFile = new File(emulatorDir, 'source.properties')
```

Comment:
Are you referring to the definition of `emulatorPropertiesFile`? It's only used in that block and no where outside of it.